### ISSUE TITLE
Cannot Create Dynamo Table

### ISSUE THREAD

**Issue by mwmisner** [alexa/skill-sample-nodejs-audio-player#6]
Hello, I was wondering if anyone had issues with dynamodb actually creating the table on invocation. I get some weird errors and would appreciate some help. (Using default stuff, followed instructions carefully)

Error

```
START RequestId: 7a2da09a-6f0c-11e6-a84c-93a4b96081e2 Version: $LATEST
2016-08-30T23:50:06.569Z    7a2da09a-6f0c-11e6-a84c-93a4b96081e2    Error during DynamoDB put:ResourceNotFoundException: Requested resource not found
2016-08-30T23:50:06.569Z    7a2da09a-6f0c-11e6-a84c-93a4b96081e2    Error saving state: ResourceNotFoundException: Requested resource not found
ResourceNotFoundException: Requested resource not found
    at Request.extractError (/var/task/node_modules/alexa-sdk/node_modules/aws-sdk/lib/protocol/json.js:43:27)
    at Request.callListeners (/var/task/node_modules/alexa-sdk/node_modules/aws-sdk/lib/sequential_executor.js:105:20)
    at Request.emit (/var/task/node_modules/alexa-sdk/node_modules/aws-sdk/lib/sequential_executor.js:77:10)
    at Request.emit (/var/task/node_modules/alexa-sdk/node_modules/aws-sdk/lib/request.js:668:14)
    at Request.transition (/var/task/node_modules/alexa-sdk/node_modules/aws-sdk/lib/request.js:22:10)
```

Test Config Data

```
{
  "session": {
    "sessionId": "SessionId.8e30439a-8d05-47c8-a132-31cdc5f2f8f0",
    "application": {
      "applicationId": "amzn1.ask.skill.Super-Secret-ID"
    },
    "attributes": {},
    "user": {
      "userId": "amzn1.ask.account.Account-secrets"
    },
    "new": true
  },
  "request": {
    "type": "IntentRequest",
    "requestId": "EdwRequestId.2e07a024-6418-4bba-a041-3e9c53eaefcc",
    "locale": "en-US",
    "timestamp": "2016-08-30T23:45:51Z",
    "intent": {
      "name": "PlayAudio",
      "slots": {}
    }
  },
  "version": "1.0"
}
```

IAM Policy

```
{
  "Version": "2012-10-17",
  "Statement": [
    {
      "Effect": "Allow",
      "Action": [
        "cloudwatch:*",
        "dynamodb:*",
        "lambda:*",
        "logs:*"
      ],
      "Resource": "*"
    }
  ]
}
```

**Comment by deegles** [comment]
If it is the first time you invoke the skill, it can take a few seconds for the table to be created. Did the table appear in DynamoDB later?

You can also create the table yourself before invoking your skill for the first time.

**Comment by mwmisner** [comment]
The API call:
I waited a couple of hours last night and never saw it. (I wonder if the invocation is so short)

From the manual creation perspective:
 Any Idea what to fill the table with? I attempted to create the table last night, but I could not figure out what indexes to use. (It would be awesome if I could get a screenshot of someones table)

**Comment by sea-steve** [comment]
I ran into the same issue trying to test from the Lambda console. When I tried it from my Echo, it worked (the audio played) and the DynamoDB table now exists.

**Comment by dreed47** [comment]
same error as others and I can't seem to get the table to be created.  and doesn't work if I manually create it.  Not sure what to use as the key either.   Can someone look at their DB and tell us what the index field is?

**Comment by sc5606** [comment]
Hello,
I am also getting the same error
"Error during DynamoDB put:ResourceNotFoundException: Requested resource not found"
Error saving state: ResourceNotFoundException: Requested resource not found
ResourceNotFoundException: Requested resource not found
at Request.extractError (/var/task/node_modules/aws-sdk/lib/protocol/json.js:43:27)
at Request.callListeners (/var/task/node_modules/aws-sdk/lib/sequential_executor.js:105:20)
at Request.emit (/var/task/node_modules/aws-sdk/lib/sequential_executor.js:77:10)
at Request.emit (/var/task/node_modules/aws-sdk/lib/request.js:668:14)
at Request.transition (/var/task/node_modules/aws-sdk/lib/request.js:22:10)
at AcceptorStateMachine.runTo (/var/task/node_modules/aws-sdk/lib/state_machine.js:14:12)
at /var/task/node_modules/aws-sdk/lib/state_machine.js:26:10
at Request.<anonymous> (/var/task/node_modules/aws-sdk/lib/request.js:38:9)
at Request.<anonymous> (/var/task/node_modules/aws-sdk/lib/request.js:670:12)
at Request.callListeners (/var/task/node_modules/aws-sdk/lib/sequential_executor.js:115:18)

No table getting created in DynamoDB. Can some one please help me creating that table manually?

**Comment by inhumantsar** [comment]
@sc5606 @dreed47 (and whoever else) If you go to https://echosim.io/ and hook it to your account, you can say "alexa, tell my-skill to play" and it will trigger table creation.

**Comment by inhumantsar** [comment]
Also, the primary key is "userId"

**Comment by sc5606** [comment]
Thank you @inhumantsar. For some reason it didn't create thru echosim.io so I ended up creating manually with userId and eagerly waiting to go home and test it out. Shall soon update with the results.

**Comment by sc5606** [comment]
@inhumantsar Thank you so much for the hint with userId column. I am able successfully play the podcast on my device with one more change which is using the Application ID from the Development step under Skill Information (replacing with this Application ID in constants.js file). 
Now, I have another question, I want to play internet radio link "http://www.onlineradios.in/manasutho-telugu/" instead of the podcast. So, I replaced the code with the link mentioned and I get an error saying it has to be with https so, I changed the link to https (which still plays on chrome browser) and I still get an error below. Can you please help me how I can play streaming audio from an url?
Thanks a bunch for your help. 
{
    "message": "Playlist Parsing Failed",
    "type": "MEDIA_ERROR_UNKNOWN"
}

**Comment by inhumantsar** [comment]
Alexa will only be able to read actual media files, but that link results in a webpage. You will need to use a link that goes directly to an audio file. eg: http://podcast.cbc.ca/mp3/podcasts/airplayyk1_20161202_34191.mp3 

A lot of streaming radio stations use M3U files for their streams. These are technically playlists but might still work.

(disclaimer: definitely not an expert)

**Comment by sc5606** [comment]
@inhumantsar I got it working my friend. I found the url for the radio stations that I wanted to add with pls and I added them to my code and it's working perfectly. I am just loving it so far... Thanks a bunch for your help with table creation. That was the major obstacle and you helped me a lot with that.
Thank you. 
Now I need to understand more with utterance's to include "what is playing" so that Alexa can tell me the title.
